src/sparsebase/preprocess/preprocess.cc

```diff
@@ -415,6 +415,44 @@ IDType *RCMReorder<IDType, NNZType, ValueType>::GetReorderCSR(
   return Q;
 }
 
+#ifdef USE_AMD_ORDER
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include <amd.h>
+#ifdef __cplusplus
+}
+#endif
+
+template <typename IDType, typename NNZType, typename ValueType>
+AMDReorder<IDType, NNZType, ValueType>::AMDReorder(AMDReorderParams p){
+  this->params_ = std::make_unique<AMDReorderParams>(p);
+  this->RegisterFunction({format::CSR<IDType, NNZType, ValueType>::get_id_static()},
+                         AMDReorderCSR);
+}
+
+template <typename IDType, typename NNZType, typename ValueType>
+IDType* AMDReorder<IDType, NNZType, ValueType>::AMDReorderCSR(std::vector<format::Format*> formats, PreprocessParams* p){
+  auto csr = formats[0]->AsAbsolute<format::CSR<IDType, NNZType, ValueType>>();
+  auto csr_converted = csr->template Convert<format::CSR, long, long, ValueType>(false);
```

Comment:
a type exception needs to be thrown if the conversion fails.

Reply:
True.